library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -375,8 +374,17 @@ private boolean updateLocked() {
         return isActive;
     }
 
+    private boolean isBatchStartingForTheFirstTime(DownloadBatch downloadBatch) {
```

Comment:
given it no longer goes to the db i imagine you can get not the most up to date data but should not be a big deal (if it actually runs this way). am i wrong?